src/View/mainpage/WindowingSlider.py

```diff
@@ -0,0 +1,297 @@
+        self.histogram_view.viewport().installEventFilter(self)
+
+        self.density = QLineSeries()
+        self.slider_density = int(self.height() / 3)
+
+        # Create sliders
+        self.sliders = QHorizontalBarSeries()
+        self.density = QLineSeries()
+        self.sliders.setLabelsVisible(False)
+        self.slider_bars = []
+        for i in range(0, self.slider_density):
+            self.slider_bars.append(QBarSet(""))
+            self.slider_bars[-1].append(1)
+            self.slider_bars[-1].setColor("white")
+            self.sliders.append(self.slider_bars[-1])
+        self.histogram.addSeries(self.sliders)
+        self.histogram.zoom(2)  # at default zoom bars don't fill the chart
+
+        # Layout
+        self.layout = QGridLayout()
+        self.layout.addWidget(self.histogram_view, 0, 0)
+        self.setLayout(self.layout)
+
+        # index of slider positions
+        self.top = self.slider_density - 1
+        self.bottom = 0
+
+        # Get the values for window and level from the dict
+        patient_dict_container = PatientDictContainer()
+        # print(patient_dict_container.get("dict_windowing"))  # testing
+        windowing_limits = patient_dict_container.get("dict_windowing")['Normal']
+
+        # Set window and level to the new values
+        window = windowing_limits[0]
+        level = windowing_limits[1]
+        self.set_bars_from_window(window, level)
+
+        # Middle drag
+        self.mouse_held = False
+        self.selected_bar = "top"
+        self.middle_drag_bounds = {
+            "top_origin": 0,
+            "bottom_origin": 0
+        }
+
+        # Test
+        self.set_density_histogram(gen_random_histogram())
+
+    def resizeEvent(self, event):
+        self.histogram.setPlotArea(
+            QtCore.QRectF(0, 0, self.fixed_width, event.size().height()))
+
+    def height_to_index(self, pos):
+        """
+        Converts graph coordinates to a slider index
+        :param pos: a local coordinate on the graph
+        """
+        index = self.slider_density-1-int(pos/(self.height()/self.slider_density))
+        if index < 0:
+            return 0
+        if index >= self.slider_density:
+            return self.slider_density-1
+        return index
+
+    def window_to_index(self, val):
+        """
+        Converts a window value to a slider index
+        :param val: a 0-2000 value
+        """
+        normalized_val = val / 2000
+        # print(normalized_val)
+        index = ceil(self.slider_density * (1 - normalized_val)) - 1
+        # print(index)
+        return index
+
+    def set_bars_from_window(self, window, level):
+        """
+        Triggered when the user selects a windowing preset.
+        Adjusts the values into the 0-2000 range, then updates the bars.
+        :param window: the window value of the preset
+        :param level: the level value of the preset
+        """
+        level = level + 1000
```

Comment:
rather than a magic value, best to assign as a constant or a variable.  If this 1000 is specifically 1/2 of the "2000", then a variable that is 1/2 of that constant or variable would be appropriate